packages/protocol/engine/eviction/state.go

```diff
@@ -117,10 +116,14 @@ func (s *State) RemoveRootBlock(id models.BlockID) {
 
 // IsRootBlock returns true if the given block is a root block.
 func (s *State) IsRootBlock(id models.BlockID) (has bool) {
+	if id == models.EmptyBlockID {
+		return true
+	}
```

Comment:
Because otherwise blocks attaching to Genesis won't be considered solid, as one of the base conditions is that parents are checked to be cointained in the root blocks set.